lang/cs/Org.Apache.REEF.IMRU.Examples/PipelinedBroadcastReduce/BroadcastSenderReduceReceiverUpdateFunction.cs

```diff
@@ -84,12 +87,52 @@ internal sealed class BroadcastSenderReduceReceiverUpdateFunction : IUpdateFunct
         /// <returns>Map input</returns>
         UpdateResult<int[], int[]> IUpdateFunction<int[], int[], int[]>.Initialize()
         {
-            for (int i = 0; i < _dim; i++)
+            if (_taskState != null && _taskState.Result != null && (_taskState.Result.HasMapInput || _taskState.Result.HasResult))
             {
-                _intArr[i] = _iterations + 1;
+                if (_taskState.Result.HasMapInput)
+                {
+                    Restore(_taskState.Result.MapInput);
+                }
+                else if (_taskState.Result.HasResult)
+                {
+                    Restore(_taskState.Result.Result);
+                }
+                _iterations = _taskState.Iterations;
+            }
+            else
+            {
+                for (int i = 0; i < _dim; i++)
+                {
+                    _intArr[i] = _iterations + 1;
```

Comment:
simply say _intArr[i] = 1; to remove confusion...thats what it will always be right?